### PR TITLE
Fix Insights ranking, MCP display names, and sync traceability

### PR DESCRIPTION
**Pull request by angadsingharora**
## Summary

Four narrow, independent UX fixes ahead of wider Decant sharing, plus the
follow-up work from review.

**Insights ranking.** The `heavy-tool` and `heavy-server` signals scored raw
`calls/4` and `calls/2`, so any high-volume built-in tool always won the hero
slot over genuinely actionable signals. Both now sit at a flat
`USAGE_SIGNAL_SCORE`, below the lowest score any issue-flagging signal can
produce, so usage volume alone never outranks a flagged problem. On a real
archive this moves the hero from `heavy-server:codex_apps` (2110 calls, which
scored 1055 under the old formula) to `ingest-health`.

**MCP server display names.** The Server column and the server filter rendered
`mcp_server` raw, showing `claude_ai_Dosu` and `plugin_posthog_posthog`.
Stripping the namespace prefix reads better but is lossy, because the same
logical server reaches the archive three ways, as a bare `.mcp.json` id, as a
claude.ai connector, and inside a plugin bundle. A real archive had six such
groups out of 33 servers, and simply prettifying the name turned them into
rows that look duplicated rather than the two real registrations they are. It
was worst where call volumes are close, `posthog` at 786 against
`plugin_posthog_posthog` at 444.

So the naming is set-aware rather than per-value. `mcpServerLabels` builds
labels across everything on screen and appends provenance only where a name is
shared, giving "Dosu (local)" against "Dosu (connector)" while every
unambiguous server keeps its short name. All 33 slugs in a real archive now
resolve to 33 distinct labels. The logic lives in `src/mcp-names.ts` so
`recommendations.ts` shares it, with `src/ui/mcp-server.ts` re-exporting to
keep `main.tsx` importing from `./` like the rest of the UI.

It is wired through both Tools and MCP tables, the server filter, the call
drawer, the Insights cards, and the standalone HTML report, so a server reads
the same everywhere. Labels come from the union of both tables because each is
limited independently. Sorting deliberately uses the short name so two
registrations of one server sort adjacently. Filter values, drilldown hrefs,
and recommendation keys keep the raw slug, and heavy-server prompts do too,
since an agent has to find the server in a config file. Two lossy edges found
in real data are fixed as well. Hyphens now split as words, so
`claude-in-chrome` reads "Claude In Chrome", a camelCase slug keeps its
capitals but gains an initial one, and a server registered under a generated
UUID is left whole instead of word-split into hex noise.

This is display only, so the `dosu` and `claude_ai_Dosu` attribution
duplication is still a separate tracked bug. Worth knowing that
`CONFIRMED_DOSU_SERVER_IDS` in `src/dosu.ts` already hardcodes that same pair,
so two mechanisms now encode the same two-registration reality by different
means.

**Sync issue traceability.** `decant sync` reported an issue count with no way
to trace it to a session. When `issues > 0` it now prints, and includes as
`issues_hint` in `--json`, a pointer to `decant ls --json` filtered on both
issue counters plus the existing `GET /api/sessions/:id/issues`. Both counters
matter because `ingest_issue_count` only counts `unparsed_line`, while
`unknown_record_type` and the other sensors land in
`informational_ingest_issue_count`. Filtering on the first alone came back
empty for exactly the scenario the new test builds. The hint prepends `--db`
when a non-default archive is in play, reading through to the real environment
so it fires for `DECANT_DB` and not only for an explicit flag, and the path is
shell-quoted so an archive under a directory with a space stays
copy-pasteable.

## Why

Decant is about to go in front of more people, and these were the four things
most likely to make it look broken on first contact. A hero insight that only
ever said "you use Bash a lot", a Server column full of internal slugs, six
sets of rows that read as duplicates once those slugs were prettified, and an
issue count with no thread to pull.

## Validation

- [x] `just check` passes (`bun test`, `bunx tsc --noEmit`, `bunx biome check .`, distribution staging smoke).
- [x] New behavior has focused tests. No existing test was weakened or deleted to make this pass.

Tests went from 847 to 868. `test/mcp-names.test.ts` is a table covering every
collision group, origin, hyphen, camelCase, and UUID case observed in a real
archive, and it supersedes `test/ui-mcp-server.test.ts` now that the logic sits
in a core module. `test/report.test.ts` gained MCP coverage, since its only
tool fixture was a builtin with a null server.

Verified against a real archive rather than only fixtures.

- All 33 server slugs resolve to 33 distinct labels, with previously colliding
  pairs sorting adjacently.
- The Insights card reads `Heavy reliance on the "Codex Apps" MCP server` while
  its prompt still names `"codex_apps"`.
- The emitted sync hint was copied verbatim and executed through a path
  containing a space, returning the one affected session. Confirmed the
  `DECANT_DB` and default-archive branches separately.
- A regression test pins the ranking relationship rather than the magic
  numbers, and the `USAGE_SIGNAL_SCORE` invariant is now asserted across every
  issue signal the fixture produces rather than only stated in a comment.

Still open on purpose. Error signal titles keep raw tool names such as
`mcp__claude_ai_Supabase__execute_sql`, because those error `detail` strings
feed the `renderedErrorServerIdentity` matcher that migrates previously marked
recommendations, so relabelling them wants its own pass with a migration test.

## Archive schema

- [x] This change does not touch the archive schema.
- [ ] This change bumps the schema. It ships the migration, and the summary above says what happens to archives created before it.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/cli.ts [hunk 1]
```diff
@@ -216,13 +216,19 @@ export async function runCli(argv: string[], options: CliRunOptions = {}): Promi
         ...archive.config,
         sourcePaths: commandOptions.path,
       });
+      const issuesHint =
+        report.issues > 0
+          ? "inspect affected sessions with: decant ls --json | jq '[.[] | select(.ingest_issue_count > 0)]' " +
```
Comment (teedole) [inline src/cli.ts]: > [!CAUTION]
> **Critical** · blocking, one line fix

Love this addition. Closing the loop from a bare issue count to the actual session is exactly what sync was missing.

There's one subtle trap in the pointer though. In `ls` output, `ingest_issue_count` only counts `unparsed_line` (real data loss). Everything else, including `unknown_record_type`, lands in `informational_ingest_issue_count` instead (lines 115 to 120 of `src/query.ts`). That split is deliberate. It's the same distinction the exit code comment below leans on, and `test/query.test.ts` pins it in the test named "session summaries separate actionable ingest failures from informational sensors".

So for the exact scenario this PR's new test creates, the suggested command comes back empty. I reproduced it on a scratch archive built from `fixtures/claude/sample.jsonl` plus one mystery record

```
sync --json          -> issues: 1 (unknown_record_type)
the hint's jq        -> []
summing both counts  -> [{ id: 1, ingest_issue_count: 0, informational_ingest_issue_count: 1 }]
```

Summing both counters makes the pointer work for every issue class

```suggestion
          ? "inspect affected sessions with: decant ls --json | jq '[.[] | select(.ingest_issue_count + .informational_ingest_issue_count > 0)]' " +
```

File: test/cli.test.ts [hunk 2]
```diff
@@ -516,10 +516,13 @@ describe("runCli", () => {
...
     );
     expect(soft.code).toBe(0);
-    expect(JSON.parse(soft.stdout)).toMatchObject({
+    const softReport = JSON.parse(soft.stdout);
+    expect(softReport).toMatchObject({
       issues: 1,
       issues_by_code: { unknown_record_type: 1 },
     });
+    expect(softReport.issues_hint).toContain("decant ls --json");
+    expect(softReport.issues_hint).toContain("/api/sessions/:id/issues");
```
Comment (teedole) [inline test/cli.test.ts]: > [!WARNING]
> **High** · goes with the hint fix

These assertions pin that the hint shows up, but not that following it finds anything. That's how the counter mix-up slipped through, since the fixture's `unknown_record_type` is exactly the class the current filter misses. Cheapest guard is to pin the corrected filter text

```suggestion
    expect(softReport.issues_hint).toContain("decant ls --json");
    expect(softReport.issues_hint).toContain("informational_ingest_issue_count");
    expect(softReport.issues_hint).toContain("/api/sessions/:id/issues");
```

If you want belt and braces, actually running the filter would keep the hint from ever drifting away from `ls` semantics again

```ts
const flagged = JSON.parse(
  (await runCli(["--db", fixtureCase.dbPath, "--json", "--no-sync", "ls"])).stdout,
).filter(
  (row: { ingest_issue_count: number; informational_ingest_issue_count: number }) =>
    row.ingest_issue_count + row.informational_ingest_issue_count > 0,
);
expect(flagged).toHaveLength(1);
```

Totally fine to stop at the string assertions if you'd rather keep this tight.

File: src/ui/main.tsx [hunk 3]
```diff
@@ -6395,6 +6395,28 @@ function formatInt(value: number): string {
...
+  if (raw == null || raw === "") {
+    return "";
+  }
+  let rest = raw;
+  for (const prefix of MCP_SERVER_PREFIXES) {
+    if (rest.startsWith(prefix)) {
+      rest = rest.slice(prefix.length);
+      break;
+    }
+  }
```
Comment (teedole) [inline src/ui/main.tsx]: > [!IMPORTANT]
> **Medium** · should fix, small

`plugin_vercel_` is one member of a family. Claude Code plugin servers all slug as `plugin_<plugin>_<server>`, so a PostHog plugin shows up as `plugin_posthog_posthog` and this renders it as Plugin Posthog Posthog. Matching the shape instead of the vendor future-proofs it without growing the list

```suggestion
const MCP_SERVER_PREFIXES = [/^claude_ai_/, /^plugin_[^_]+_/];

function formatMcpServer(raw: string | null | undefined): string {
  if (raw == null || raw === "") {
    return "";
  }
  let rest = raw;
  for (const prefix of MCP_SERVER_PREFIXES) {
    if (prefix.test(rest)) {
      rest = rest.replace(prefix, "");
      break;
    }
  }
```

With that, `plugin_vercel_vercel` still gives Vercel, `plugin_posthog_posthog` gives Posthog, and hyphenated plugin names keep working since the wildcard stops at the underscore.

Related thought, take it or leave it. This is a pure formatter with real edge cases (null, empty string, a prefix-only value like `claude_ai_` currently formats to an empty cell), and the repo's pattern for exactly this shape is an extracted `src/ui/<module>.ts` plus a `test/ui-<module>.test.ts` table, like `date-time.ts` or `tool-filters.ts`. The manual verification table from the PR description would make a great unit test, and it would have caught the posthog case.

File: src/ui/main.tsx [hunk 4]
```diff
@@ -7234,7 +7256,7 @@ function ToolsView({
                       <td className="mono">
                         <span className="icon-cell drilldown-label">
                           <Icon name="cpu" />
-                          <span>{row.mcp_server}</span>
+                          <span>{formatMcpServer(row.mcp_server)}</span>
```
Comment (teedole) [inline src/ui/main.tsx]: > [!NOTE]
> **Low** · nice to have

With `dosu` and `claude_ai_Dosu` both formatting to Dosu (the attribution dup you already flagged as tracked separately), the two rows go from awkwardly duplicated to indistinguishable until that fix lands. A title attribute keeps them tellable apart on hover, and doubles as a home for the raw slug for anyone who greps by it

```suggestion
                          <span title={row.mcp_server}>{formatMcpServer(row.mcp_server)}</span>
```

Same tweak would fit the tool table cell around line 7382. And while you're in this row, the aria label a couple lines up still reads the raw slug. Now that the visible text is formatted, running it through `formatMcpServer` there too keeps the accessible name matching what sighted users see.

File: src/ui/main.tsx [hunk 5]
```diff
@@ -7415,7 +7437,7 @@ function ToolsView({
               <option value="">All servers</option>
               {data.mcp.map((row) => (
                 <option key={row.mcp_server} value={row.mcp_server}>
-                  {row.mcp_server} ({formatInt(row.calls)})
+                  {formatMcpServer(row.mcp_server)} ({formatInt(row.calls)})
```
Comment (teedole) [inline src/ui/main.tsx]: > [!NOTE]
> **Low** · nice to have

Same collision here, and it bites hardest in a dropdown. Two literally identical Dosu options that differ only in their hidden value. A title is cheap insurance

```suggestion
                <option key={row.mcp_server} title={row.mcp_server} value={row.mcp_server}>
                  {formatMcpServer(row.mcp_server)} ({formatInt(row.calls)})
```

File: src/ui/main.tsx [hunk 6]
```diff
@@ -7357,7 +7379,7 @@ function ToolsView({
                       {row.mcp_server != null && row.mcp_server !== "" ? (
                         <span className="icon-cell">
                           <Icon name="cpu" />
-                          <span>{row.mcp_server}</span>
+                          <span>{formatMcpServer(row.mcp_server)}</span>
```
Comment (teedole) [inline src/ui/main.tsx]: > [!IMPORTANT]
> **Medium** · should fix, small

This one has no apply button since the fix lands just outside the diff.

Now that the Server cells render formatted names, the column sort still compares raw slugs (the `case "server"` branches in `mcpSortValue` and `toolSortValue`, around lines 3369 and 3388). Sorting ascending can render as Dosu, Codex Apps, Dosu, Vercel, which looks broken to anyone who clicks the header. Pointing both branches at the formatter lines it back up

```ts
    case "server":
      return formatMcpServer(row.mcp_server);
```

File: src/recommendations.ts [hunk 7]
```diff
@@ -691,7 +691,11 @@ function heavyServers(mcp: ReturnType<typeof mcpUsage>): Recommendation[] {
             icon: "hero-cpu-chip",
             tone: "accent",
             impact_label: `${server.calls} calls`,
-            score: server.calls / 2,
+            // Flat and low: volume alone isn't a problem, so this must lose to
+            // every signal that flags an actual issue. The lowest score any of
+            // those can produce is errorHotspots' eligibility floor (20 calls at
+            // a 12% error rate = 2.4), so this stays under that.
+            score: 2,
```
Comment (teedole) [inline src/recommendations.ts]: > [!NOTE]
> **Low** · optional

The change itself looks right to me, so treat this one as optional hardening.

I really like that the reasoning is written down here, and I checked it. With today's constants every issue-flagging signal floors above 2 (error 2.4, ingest health 4, cost 5, hot context 8, abandoned about 8.3, churn 9, search heavy 10). The fragile part is that the invariant lives in a comment while its inputs live in formulas and constants scattered across the file. One concrete drift scenario. Drop `INGEST_HEALTH_MIN_SESSIONS` from 15 to 10 and a single affected session scores exactly 2, and since the global tie break is key-ascending (line 111), `heavy-` beats `ingest-health` on the tie. Two cheap ways to make the invariant load-bearing, either or both. A named `USAGE_SIGNAL_SCORE = 2` next to the thresholds at the top of the file, and a small test asserting each issue signal's minimum construction outscores it.

Tiny heads-up, not a request. Flat scores also mean the heavy entries now order alphabetically among themselves rather than by volume, so 85 calls can sit above 250 calls in the list. If that ever looks odd, `2 - 1 / calls` preserves volume order inside the band, at the cost of loosening the exact score expectations in the test. Genuinely only if it bothers anyone.

File: test/recommendations.test.ts [hunk 8]
```diff
@@ -147,6 +147,13 @@ describe("recommendations", () => {
       tone: "warning",
       impact_label: "83% of spend",
     });
+    // The Insights UI picks its "hero" as the highest-scored open signal.
+    // A tool being heavily used is not itself a problem, so it must never
+    // outrank a signal that flags a real issue -- like this 20% error rate.
+    const usageVsError = rows
+      .filter((row) => row.key.startsWith("signal:heavy-") || row.key === error?.key)
+      .sort((left, right) => right.score - left.score);
+    expect(usageVsError[0]?.key).toBe(error?.key);
```
Comment (teedole) [inline test/recommendations.test.ts:159]: > [!TIP]
> 🎉 **Kudos** · much appreciated

This is exactly the right shape of regression test. It pins the ranking relationship instead of the magic numbers, so it'll survive future score tuning. Nice.

**Review by teedole — CHANGES_REQUESTED** [review]
Thanks @angadsingharora, these are exactly the right three cuts ahead of sharing Decant more widely, and each one stays neatly in its lane. The hint lives in `cli.ts` per invariant 1, the formatting is display-only with raw filter values and hrefs, and there's no schema or API contract impact since `POST /api/sync` builds its report on a separate path. Folding in the gemini null guard was a good call too.

I'm requesting changes for exactly one thing. The jq pointer in the sync hint filters on the wrong counter and comes back empty for the very scenario the new test builds. Details and a one line suggestion are inline. Everything else opens with a priority callout, and the medium and low ones are take it or leave it. Take what resonates and push back where you disagree.

A few small things with no diff line to hang them on

- The tool detail drawer eyebrow (`main.tsx` around line 6891, `call.mcp_server ?? call.tool_kind`) still shows the raw slug. If you point it at `formatMcpServer`, mind that the formatter returns an empty string rather than null, so the `??` chain won't fall through on its own.
- The Insights card titled Heavy reliance on the `claude_ai_Dosu` MCP server composes its title server-side in `recommendations.ts`, so it will show the raw slug right next to a freshly formatted Tools and MCP view. Fine for now, just flagging it so the inconsistency is a decision rather than a surprise.
- The text-mode hint line is only exercised through `--json`. A one line stderr assertion on a non-quiet text run would cover the other half.
- One copy-paste nit. The hint always says `decant ls --json`, so anyone running with `--db` or `DECANT_DB` pointed at a non-default archive will inspect the wrong one. Prepending the flag when it's set would be a nice touch, also fine to skip.

Happy to pair on any of these. Excited to get this in front of more people.

**Review by teedole — APPROVED** [review]
LGTM 👍🏼
